app/org/sagebionetworks/bridge/services/email/ConsentEmailProvider.java

```diff
@@ -107,17 +111,47 @@ public MimeTypeEmail getEmail(String defaultSender) throws MessagingException {
         return builder.build();
     }
 
+    /**
+     * Consent documents were originally whole XHTML documents (they must be valid XML because PDF support from Java
+     * libraries is limited and bad, and the tool we're using only works with XML as an input). In order to edit these
+     * documents, we are moving to a system where only the content portion of the consent document, excluding the
+     * signature block at the end, is available to researchers to edit. We then assemble the complete HTML document at
+     * runtime. Here we branch based on the detection of a complete HTML document and do either one or the other.
+     * 
+     * @return
+     */
     private String createSignedDocument() {
         StudyConsentView consent = studyConsentService.getActiveConsent(study.getStudyIdentifier());
         String consentAgreementHTML = consent.getDocumentContent();
-        
         String signingDate = FORMATTER.print(DateUtils.getCurrentMillisFromEpoch());
-        String html = consentAgreementHTML.replace("@@name@@", consentSignature.getName());
-        html = html.replace("@@signing.date@@", signingDate);
-        html = html.replace("@@email@@", user.getEmail());
         String sharingLabel = (sharingScope == null) ? "" : sharingScope.getLabel();
-        html = html.replace("@@sharing@@", sharingLabel);
-        return html;
+        
+        if (consentAgreementHTML.contains("<html")) {
+            // proceed as we used to
+            String html = consentAgreementHTML.replace("@@name@@", consentSignature.getName());
+            html = html.replace("@@signing.date@@", signingDate);
+            html = html.replace("@@email@@", user.getEmail());
+            html = html.replace("@@sharing@@", sharingLabel);
+            return html;
+        } else {
+            // This is now a fragment, assemble accordingly
+            Map<String,String> map = Maps.newHashMap();
+            map.put("studyName", study.getName());
+            map.put("supportEmail", study.getSupportEmail());
+            map.put("technicalEmail", study.getTechnicalEmail());
+            map.put("sponsorName", study.getSponsorName());
+            String resolvedConsentAgreementHTML = BridgeUtils.resolveTemplate(consentAgreementHTML, map);
+
+            map = Maps.newHashMap();
+            map.put("studyName", study.getName());
+            map.put("consent.body", resolvedConsentAgreementHTML);
```

Comment:
Since "consent.body" is checked as a regex, I wonder if you want to escape the dot.